File: 387-first-unique-character-in-a-string/387-first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        freq = Counter(s);
        # for ch in s:
        #     if ch in freq:
        #         freq[ch] += 1;
        #     else:
        #         freq[ch] = 1;
        i = 0;
        for ch in s:
            if 1==freq[ch]:
                return i;
            i += 1;
        return -1;

File: 387-first-unique-character-in-a-string/387-first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        vector<int> freq(26,0);
        for(char ch : s){
            freq[ch-'a']++;
        }
        int i=0;
        for(char ch : s){
            if(1==freq[ch-'a'])
                return i;
            i++;
        }
        return -1;
    }
};